core/src/main/java/org/hisp/dhis/android/core/tracker/exporter/TrackerDownloadCall.kt

```diff
@@ -50,88 +55,85 @@ import org.hisp.dhis.android.core.user.internal.UserOrganisationUnitLinkStore
 
 @Suppress("TooManyFunctions")
 internal abstract class TrackerDownloadCall<T, Q : BaseTrackerQueryBundle>(
-    private val rxCallExecutor: RxAPICallExecutor,
+    private val coroutineAPICallExecutor: CoroutineAPICallExecutor,
     private val userOrganisationUnitLinkStore: UserOrganisationUnitLinkStore,
     private val systemInfoModuleDownloader: SystemInfoModuleDownloader,
     private val relationshipDownloadAndPersistCallFactory: RelationshipDownloadAndPersistCallFactory
 ) {
 
-    fun download(params: ProgramDataDownloadParams): Observable<TrackerD2Progress> {
-        return Observable.defer {
-            val progressManager = TrackerD2ProgressManager(null)
-            if (userOrganisationUnitLinkStore.count() == 0) {
-                return@defer Observable.fromCallable {
-                    progressManager.setTotalCalls(1)
-                    progressManager.increaseProgress(TrackedEntityInstance::class.java, true)
-                }
-            } else {
-                val relatives = RelationshipItemRelatives()
-                return@defer systemInfoModuleDownloader.downloadWithProgressManager(progressManager)
-                    .switchMap {
-                        rxCallExecutor.wrapObservableTransactionally(
-                            Observable.merge(
-                                Observable.defer { downloadInternal(params, progressManager, relatives) },
-                                Observable.defer { downloadRelationships(progressManager, relatives) },
-                                Observable.fromCallable { progressManager.complete() }
-                            ),
-                            cleanForeignKeys = true
+    @OptIn(ExperimentalCoroutinesApi::class)
+    fun download(params: ProgramDataDownloadParams): Flow<TrackerD2Progress> = flow {
```

Comment:
Here is the issue, ` coroutineAPICallExecutor.wrapTransactionally` now is waiting to execute the block of flows (merged in a single flow) and then it finish the transaction, but it seems that the collect is not emiting, at least in the test in [EventDownloadCallMockIntegrationShould.kt](https://github.com/dhis2/dhis2-android-sdk/pull/1986/files#diff-66a45112c1eed316d562d8f4d713240f8c475bdebae91e0cb244f7bb14229412) is failing